### DISCUSSION TITLE
react-native repo to include FlexLayout source for experimentation

### DISCUSSION THREAD

**Discussion by JTYim** [react-native-community/discussions-and-proposals#499] [Ideas]
In the near future, the react-native repo will be exposing some of the source for the FlexLayout layout engine for exploratory purposes. We're currently experimenting with this alternative layout engine for Yoga, which we hope might help to address some of the pain points brought by the community over the years.

This inclusion of the new files in the repository can be safely ignored as they will have no functional impact on end users during this phase, ie. APK size will not be affected, and the layout API and functionality will remain entirely unchanged. 

Please also note that we also will not yet be considering feature or pull requests for the FlexLayout engine at this point in time. Feel free to comment on this discussion if you have any concerns or questions related to FlexLayout.

**Comment by Titozzz** [comment]
Related to https://cocoapods.org/pods/FlexLayout ?

**Reply by NickGerleman** [reply]
Different project, with coincidentally the same name it looks like 😅. We will have to think about reconciling that...

**Reply by JTYim** [reply]
As @NickGerleman said, the two share a name only. FlexLayout (the one we are exploring) has not previously been open sourced

**Comment by EvanBacon** [comment]
This looks really promising! Based on the source here https://github.com/facebook/react-native/commit/4eec47344e25b2e2cb2e6dbdb3f989ed5b6e2747 it looks like FlexLayout is cross-platform C++.

Any particular reason why this is a new package and not rolled into Yoga?

**Reply by NickGerleman** [reply]
`flexlayout` and `yoga` have roughly the same functionality, but are different implementations, with different interfaces. We are also hoping to use it as a vehicle to move towards more compliant flexbox behavior, consistent with web.

**Reply by JTYim** [reply]
I don’t want to speak for the authors of FlexLayout, and we are still very early in the experimentation phase, but echoing what @NickGerleman said. There are similarities in the core algorithm, but notable differences in the data models, core API and overall implementation.

**Reply by iainsmith** [reply]
@NickGerleman / @JTYim . Is FlexLayout now open source?

**Reply by iainsmith** [reply]
In case anyone else comes across this. This was open sourced into the react-native repo in [this commit](https://github.com/facebook/react-native/commit/4eec47344e25b2e2cb2e6dbdb3f989ed5b6e2747) and then removed in [0.7.1](https://github.com/facebook/react-native/commit/bf05df17230e1063ff29d5ae5d86feac85e122b3)

**Comment by NickGerleman** [comment]
Hey folks! I want to give a quick update on this.

After investigating integration of FlexLayout and react-native, we have found some potential long-term benefits, but a migration effort doesn’t yet make sense compared to other work we could be doing to improve layout capabilities and conformance.

Yoga is an essential part of React Native and its community, so we want to take a more active role in the Yoga repo. Within the React Native team at Meta, we don't have the same level of expertise as the original author, but part of our decision to stick with Yoga in the near-term is motivated by a desire to do better for the OSS community.

**Reply by nandorojo** [reply]
Thanks for the update @NickGerleman! I arrived here from your comment on the flex gap PR. That PR might be the most excited I've been for an update to React Native. Happy to see the attention paid towards the community lately.
